frontend/public/components/utils/dropdown.jsx

```diff
@@ -598,6 +532,43 @@ export class Dropdown extends DropdownMixin {
   }
 }
 
+export const Dropdown = (props) => {
+  const { storageKey } = props;
+
+  // Should be undefined so that we don't save undefined-bookmarks.
+  const bookmarkStorageKey = storageKey ? `${storageKey}-bookmarks` : undefined;
+
+  const [favoriteKey, setFavoriteKey] = useUserSettingsCompatibility(
+    storageKey,
+    storageKey,
+    null,
+    // TODO: enable sync when sync PR is merged
```

Comment:
@jerolimov how would the returned state work here. Since the key is undefined and we don't want to initialize it in user settings.

Reply:
I'm guessing similar to `useState`?